### PR TITLE
Refactor unitCalc without using eval to support SSR/SSG

### PR DESCRIPTION
**Pull request by jpeterson**
## Description
This PR refactors `unitCalc` so that it doesn't need to rely on `window` or `eval` to function properly. This makes it isomorphic and conducive to SSR/SSG.

**kudos to @hogpilot for recommending the logic update here 🙏🏼 🎉** 

## Related Issue
#396

## Motivation and Context
Calcite React mostly works with SSR/SSG frameworks like Next.js or Gatsby, but when `unitCalc` gets executed server-side, it returns incorrect values (see #396 for more details).

## How Has This Been Tested?
The following sandbox is forked from the repro sandbox for #396 but is using the updated `unitCalc` function.

https://codesandbox.io/s/silent-grass-mug16?file=/pages/index.js

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: src/utils/helpers.js [hunk 1]
```diff
@@ -13,27 +13,33 @@ import { useEffect, useState } from 'react';
...
+    case '-':
+      return left - right + unit;
+    case '*':
+      return left * right + unit;
+    case '/':
+      return left / right + unit;
+    case '%':
+      return (left % right) + unit;
+    default:
+      return undefined;
```
Comment (jpeterson) [inline src/utils/helpers.js:41]: @codylawson what do you think the default case should return here? Should it `console.warn` in addition to returning something?

**Comment by jpeterson** [comment]
@codylawson there are a lot more operators we could cover here if we want... but my sense is that this helper method is intended to be used for simple math operations (+, -, *, /, %). Do you think it would be at all useful to include additional JS operators?

For example, see the commented cases below:

```js
  switch (operator) {
    case '+':
      return left + right + unit;
    case '-':
      return left - right + unit;
    case '*':
      return left * right + unit;
    case '/':
      return left / right + unit;
    case '%':
      return (left % right) + unit;
    // case '==':
    //   return left == right + unit;
    // case '!=':
    //   return left != right + unit;
    // case '===':
    //   return left === right + unit;
    // case '!==':
    //   return left !== right + unit;
    // case '<':
    //   return left < right + unit;
    // case '<=':
    //   return left <= right + unit;
    // case '>':
    //   return left > right + unit;
    // case '>=':
    //   return left >= right + unit;
    // case '<<':
    //   return left << (right + unit);
    // case '>>':
    //   return left >> (right + unit);
    // case '>>>':
    //   return left >>> (right + unit);
    // case '**':
    //   return left ** right + unit;
    // case '|':
    //   return left | (right + unit);
    // case '^':
    //   return left ^ (right + unit);
    // case '&':
    //   return left & (right + unit);
    // case 'in':
    //   return left in right + unit;
    // case 'instanceof':
    //   return left instanceof right + unit;
  }
```

File: src/utils/helpers.js (hunk 1 above)
Comment (codylawson) [reply src/utils/helpers.js:41]: Yeah I like the idea of maybe just returning `left` and a `console.warn` in the default.

Comment (jpeterson) [reply src/utils/helpers.js:41]:
```suggestion
      console.warn('unitCalc only supports the following operators: +, -, *, /, %');
      return left;
```

Comment (codylawson) [reply src/utils/helpers.js:41]: LGTM

**Pull request merged**